airbyte-ci/connectors/live-tests/src/live_tests/commons/models.py

```diff
@@ -0,0 +1,235 @@
+        if self.command is Command.DISCOVER:
+            self.raise_if_missing_attr_for_command("config")
+        if self.command is Command.READ:
+            self.raise_if_missing_attr_for_command("config")
+            self.raise_if_missing_attr_for_command("catalog")
+        if self.command is Command.READ_WITH_STATE:
+            self.raise_if_missing_attr_for_command("config")
+            self.raise_if_missing_attr_for_command("catalog")
+            self.raise_if_missing_attr_for_command("state")
+
+
+@dataclass
+class ExecutionResult:
+    stdout: str
+    stderr: str
+    executed_container: dagger.Container
+    http_dump: Optional[dagger.File]
+    airbyte_messages: List[AirbyteMessage] = field(default_factory=list)
+    airbyte_messages_parsing_errors: List[Tuple[Exception, str]] = field(default_factory=list)
+
+    def __post_init__(self):
+        self.airbyte_messages, self.airbyte_messages_parsing_errors = self.parse_airbyte_messages_from_command_output(self.stdout)
+
+    @staticmethod
+    def parse_airbyte_messages_from_command_output(
+        command_output: str,
+    ) -> Tuple[List[AirbyteMessage], List[Tuple[Exception, str]]]:
+        airbyte_messages: List[AirbyteMessage] = []
+        parsing_errors: List[Tuple[Exception, str]] = []
+        for line in command_output.splitlines():
+            try:
+                airbyte_messages.append(AirbyteMessage.parse_raw(line))
+            except ValidationError as e:
+                parsing_errors.append((e, line))
+        return airbyte_messages, parsing_errors
+
+
+@dataclass
+class ExecutionReport:
+    execution_inputs: ExecutionInputs
+    execution_result: ExecutionResult
+    created_at: int = field(default_factory=lambda: int(time.time()))
+
+    @property
+    def report_dir(self) -> str:
+        return f"{self.created_at}/{self.execution_inputs.connector_under_test.name}/{self.execution_inputs.command.value}/{self.execution_inputs.connector_under_test.version}/"
+
+    @property
+    def stdout_filename(self):
+        return "stdout.log"
+
+    @property
+    def stderr_filename(self):
+        return "stderr.log"
+
+    @property
+    def http_dump_filename(self):
+        return "http_dump.mitm"
+
+    async def save_to_disk(self, output_dir: Path) -> None:
+        final_dir = output_dir / self.report_dir
+        final_dir.mkdir(parents=True, exist_ok=True)
+        stdout_file_path = final_dir / self.stdout_filename
+        stdout_file_path.write_text(self.execution_result.stdout)
+
+        stderr_file_path = final_dir / self.stderr_filename
+        stderr_file_path.write_text(self.execution_result.stderr)
+        if self.execution_result.http_dump:
+            http_dump_file_path = final_dir / self.http_dump_filename
+            await self.execution_result.http_dump.export(str(http_dump_file_path.resolve()))
+        # TODO merge ExecutionReport.save_to_disk and Backend.write?
+        # Make backends use customizable
```

Comment:
I'm curious to know what you have in mind here.

Reply:
I basically  realized that to classes perform final writes to file system of artifact, could be better to have this centralized.